src/main/java/ladder/model/NameList.java

```diff
@@ -49,11 +50,13 @@ public List<String> getNameList() {
         return nameList;
     }
 
-    public void swapPoints(int index) {
-        if (index < 0 || index > nameList.size()) {
+    public NameList swapPoints(int index) {
+        if (index < 0 || index >= nameList.size() - 1) {
             throw new IllegalArgumentException("경계 초과");
         }
-        List<String> swapList = this.nameList;
-        Collections.swap(swapList, index, index + 1);
+        List<String> newList = new ArrayList<>(nameList);
+        Collections.swap(newList, index, index + 1);
+        return new NameList(String.join(",", newList));
```

Comment:
현 시점에선 Input과 아무런 관련이 없음에도 콤마로 구분자로 String 타입을 만들어야 하네요.
그냥 newList 로 NameList를 만들면 어떤가요?
```suggestion
        return new NameList(newList);
```